test/TestCase.cpp

```diff
@@ -139,3 +138,37 @@ EVMVersionRestrictedTestCase::EVMVersionRestrictedTestCase(std::string const& _f
 	if (!comparisonResult)
 		m_shouldRun = false;
 }
+
+void EVMVersionRestrictedTestCase::processCompileToFlag()
+{
+	std::optional<uint8_t> eofVersion = solidity::test::CommonOptions::get().eofVersion();
+
+	std::string compileToString = m_reader.stringSetting("bytecodeFormat", "any");
+	if (compileToString == "any" || compileToString == "legacy,>=1" || compileToString == ">=1,legacy")
+		return;
+
+	if (!m_shouldRun)
+		return;
+
+	// TODO: This is naive implementation because for now we support only one EOF version.
+	if (compileToString == "legacy" && eofVersion.has_value())
+		m_shouldRun = false;
+	else if (compileToString == ">=1")
+	{
+		// TODO: Update if EOF moved to Osaka
+		// EOF only available since Prague
+		solAssert(solidity::test::CommonOptions::get().evmVersion() >= langutil::EVMVersion::prague());
+		// Do not run if non-EOF test run
+		if (!eofVersion.has_value())
+			m_shouldRun = false;
+	}
+	else
+		BOOST_THROW_EXCEPTION(std::runtime_error{"Invalid bytecodeFormat flag: \"" + compileToString + "\""});
+}
+
+EVMVersionRestrictedTestCase::EVMVersionRestrictedTestCase(std::string const& _filename):
+	TestCase(_filename)
+{
+	processEVMVersionFlag();
+	processCompileToFlag();
```

Comment:
Naming here and in `processCompileToFlag()` should be updated to match the new setting name.

Also, I'd call these "settings" rather than "flags" since they're not `true`/`false`.

```suggestion
	processEVMVersionSetting();
	processBytecodeFormatSetting();
```